File: python practice/20260613_Rails.py
import sys

def solve():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    iterator = iter(input_data)
    
    while True:
        try:
            n_str = next(iterator)
        except StopIteration:
            break
        n = int(n_str)
        if n == 0:
            break
        
        while True:
            first_val = int(next(iterator))
            if first_val == 0:
                print()
                break
            
            target = [first_val]
            for _ in range(n - 1):
                target.append(int(next(iterator)))
            
            stack = []
            current = 1
            possible = True
            
            for coach in target:
                while current <= coach:
                    stack.append(current)
                    current += 1
                if stack and stack[-1] == coach:
                    stack.pop()
                else:
                    possible = False
                    break
            
            if possible and not stack:
                print("Yes")
            else:
                print("No")

if __name__ == '__main__':
    solve()

File: C Language/20260613_Rails.c
#include <stdio.h>

int main() {
    int n;
    while (scanf("%d", &n) == 1 && n != 0) {
        while (1) {
            int target[1005];
            scanf("%d", &target[0]);
            if (target[0] == 0) {
                printf("\n");
                break;
            }
            for (int i = 1; i < n; i++) {
                scanf("%d", &target[i]);
            }

            int stack[1005], top = -1;
            int curr = 1, possible = 1, target_idx = 0;

            while (target_idx < n) {
                if (top >= 0 && stack[top] == target[target_idx]) {
                    top--;
                    target_idx++;
                } else if (curr <= n) {
                    stack[++top] = curr++;
                } else {
                    possible = 0;
                    break;
                }
            }

            printf("%s\n", possible ? "Yes" : "No");
        }
    }
    
    return 0;
}